search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt

```diff
@@ -149,46 +149,54 @@ class TemporalEntityHandler(
 }
 
 internal fun buildTemporalQuery(params: MultiValueMap<String, String>): TemporalQuery {
-    if (!params.containsKey("timerel") || !params.containsKey("time"))
-        throw BadRequestDataException("'timerel and 'time' request parameters are mandatory")
-
-    if (params.getFirst("timerel") == "between" && !params.containsKey("endTime"))
+    val timerelParam = params.getFirst("timerel")
+    val timeParam = params.getFirst("time")
+    val endTimeParam = params.getFirst("endTime")
+    val timeBucketParam = params.getFirst("timeBucket")
+    val aggregateParam = params.getFirst("aggregate")
+    val lastNParam = params.getFirst("lastN")
+    val attrsParam = params.getFirst("attrs")
+
+    if (timerelParam == "between" && endTimeParam == null)
         throw BadRequestDataException("'endTime' request parameter is mandatory if 'timerel' is 'between'")
 
-    val timerel = try {
-        TemporalQuery.Timerel.valueOf(params.getFirst("timerel")!!.toUpperCase())
-    } catch (e: IllegalArgumentException) {
-        throw BadRequestDataException("'timerel' is not valid, it should be one of 'before', 'between', or 'after'")
+    val (timerel, time) = if (timerelParam == null && timeParam == null) {
+        Pair(null, null)
+    } else if (timerelParam != null && timeParam != null) {
+        val timeRelTemp = try {
+            TemporalQuery.Timerel.valueOf(timerelParam.toUpperCase())
+        } catch (e: IllegalArgumentException) {
+            throw BadRequestDataException("'timerel' is not valid, it should be one of 'before', 'between', or 'after'")
+        }
+        Pair(timeRelTemp, timeParam.parseTimeParameter("'time' parameter is not a valid date"))
```

Comment:
Maybe you can also avoid declaring this temp variable.

Reply:
I think it will be less readable